cookbooks/cdo-ruby/recipes/rubygems.rb

```diff
@@ -16,3 +16,29 @@
     not_if "which gem && gem --version | grep -q '#{node['cdo-ruby']['rubygems_version']}'"
   end
 end
+
+# Clean up no-longer-used root-installed gems.
+#
+# For context: we previously would install all gems as the root user for our
+# deployed web application, but with
+# github.com/code-dot-org/code-dot-org/pull/66536 we are now using bundle's
+# deployment mode to install them just for our applicaiton.
+#
+# This command removes all gems installed by the root user and their associated
+# executables, leaving only the native gems that come installed by default.
+#
+# TODO infra: remove this once it has executed successfully on all persistent
+# managed servers.
+execute 'gem uninstall --all --ignore-dependencies --executables' do
+  only_if do
+    # We only want to attempt this if we have gems installed at the target
+    # locations for both the root and the deployment installation methods.
+    #
+    # For simplicity, just check for one arbitrary gem. It's a fragile
+    # strategy, but this code is intended to be temporary and a false negative
+    # is harmless.
+    canary = '3.1.0/gems/rails-6.1.7.7'
```

Comment:
I'm confused to see the `rails` used as the canary gem since it doesn't appear in the list of gems that would be uninstalled in the PR description, but I might have misunderstood something.

Reply:
I could probably have been more clear with what I meant by that.

Right now when you spin up a fresh adhoc, it installs all of our Gemfile dependencies including Rails to `~/adhoc/vendor`. But the base AMI we use comes with I think 92 gems already installed to `/usr/local/lib`. Most of those gems remain installed even after running `gem uninstall --all`, but the ones listed above get removed.